Assets/Mapzen/PolylineBuilder.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using Mapzen.VectorData;
+using UnityEngine;
+
+namespace Mapzen
+{
+    public class PolylineBuilder : IGeometryHandler
+    {
+        public struct Options
+        {
+            public Material Material;
+            public bool Extrude;
+            public float MinHeight;
+            public float MaxHeight;
+            public float Width;
+        }
+
+        public PolylineBuilder(MeshData outputMeshData, Options options)
+        {
+            this.options = options;
+
+            var polygonOptions = new PolygonBuilder.Options();
+            polygonOptions.Material = options.Material;
+            polygonOptions.Extrude = options.Extrude;
+            polygonOptions.MinHeight = options.MinHeight;
+            polygonOptions.MaxHeight = options.MaxHeight;
+
+            polygonBuilder = new PolygonBuilder(outputMeshData, polygonOptions);
+        }
+
+        PolygonBuilder polygonBuilder;
+
+        Options options;
+
+        Vector2 lastPoint;
+        int pointsInLineString;
+
+        public void OnPoint(Point point)
+        {
+            var currPoint = new Vector2(point.X, point.Y);
+            if (pointsInLineString > 0)
+            {
+                // Start a polygon for the segment from the last point in the linestring to this one.
+                polygonBuilder.OnBeginPolygon();
+                polygonBuilder.OnBeginLinearRing();
+
+                // Create a quad around the segment.
+                var perp = Perp(-currPoint + lastPoint) * (options.Width * 0.5f);
+                polygonBuilder.OnPoint(new Point(currPoint.x - perp.x, currPoint.y - perp.y));
+                polygonBuilder.OnPoint(new Point(currPoint.x + perp.x, currPoint.y + perp.y));
+                polygonBuilder.OnPoint(new Point(lastPoint.x + perp.x, lastPoint.y + perp.y));
+                polygonBuilder.OnPoint(new Point(lastPoint.x - perp.x, lastPoint.y - perp.y));
+
+                // Repeat the last point to form a ring.
+                polygonBuilder.OnPoint(new Point(currPoint.x - perp.x, currPoint.y - perp.y));
+
+                // Finish the polygon.
+                polygonBuilder.OnEndLinearRing();
+                polygonBuilder.OnEndPolygon();
+            }
+
+            lastPoint = currPoint;
+            pointsInLineString++;
+        }
+
+        public void OnBeginLineString()
```

Comment:
You are correct, I originally had all of these returning `bool` to indicate whether the geometry producer should continue (`true`) or stop completely (`false`). However when I implemented this on the producer side in `GeoJsonFeature` and `MvtFeature` it got pretty complicated. Needing to be able to exit the decoding at any point leads to a lot of branching and makes the code harder to follow. Since I couldn't think of a concrete use case that needed this, I chose to "keep it simple, stupid" and cut it out.

If we find ourselves with a concrete need for this in the future I'm open to adding it back.